### ISSUE TITLE
Collaborate & Integrate this with "Files" app on windows

### ISSUE THREAD

**Issue by enygmator** [JasonStein/Notepads#737]
## PROBLEM
This is probably a big (and difficult) feature request, but I, and many others are still awaiting the "Sets" feature on windows which was in preview for a short period of time a few years ago and has the ability to have a "file explorer" tab and "Notepad" tab in the same window. I have an idea - we can replicate this functionality as it's really helpful.

## Solution
The "Files" app on windows (found [here](https://github.com/files-community/Files) on GitHub) is a great multi-tab windows 10 file explorer alternative, that's built by the community and is open-source too. My Idea is to collaborate with the "files" app community and try and integrate this "notepads" app into that app. They already have a multi-tab system like this app and all that needs to be done is to get an instance of notepad in-built into that app, or add the ability to install it separately as an additional feature to the "files" app (Microsoft store allows distribution of extra feature packs for apps in UWP).

This way all notepad files can be opened within the "Files" app without having to switch between windows. I am not sure, but I also think that the "files" app has access to system files, thus enabling system file support to notepads.

Or maybe, both the apps can be combined to create a separate app, so as to let users choose what they want to install.

## Misc.
I hope that this is not a duplicate issue (I checked for duplicates).  
I am not familiar with the code structure of "Notepads" and thus I don't know how feasible this is. I have been writing UWP apps, (NOT very advanced ones) for the past 4 years and I would love to work on this integration with the "Notepads" community, if this collaboration with the "files" community is taken up.

P.S.: I am using both apps as an almost permanent alternative to their system (windows in-built) counterparts. Thanks to this and the "files" community!

**Comment by JasonStein** [comment]
There is no eazy way to achive your feature requeset here unless we put the hole notepads code into files-uwp which I don't think it is a good idea. I don't think there will be a good way to do it unless Windows 10 is having the "set" feature in the future.

**Comment by enygmator** [comment]
Of course, putting the entire "notepads" code in the "files" app could cause lots of problems.

But I believe I MAY have found a way around it using a very simple technique: 
We could partition the "notepads" code such that the "text editor" part is separate from the "multiple tab systems" part and then, implement the "text editor" part as a Nuget package (with UI controls). Refer UWP - Nuget docs : https://docs.microsoft.com/en-us/nuget/guides/create-ui-controls
Then, the "notepads" app and the "files" app can both use the "text editor" as a Nuget package - UI control.

I believe that this work would be really simple to accomplish. Though, I might add that the settings for the "Text editor" may have to either be implemented as another "Nuget package UI control" or bundled with the "text editor" UI control.
What do you think?

**Comment by enygmator** [comment]
Of course, putting the entire "notepads" code in the "files" app could cause lots of problems.

But I believe I MAY have found a way around it using a very simple technique: 
We could partition the "notepads" code such that the "text editor" part is separate from the "multiple tab systems" part and then, implement the "text editor" part as a Nuget package (with UI controls). Refer []WP - Nuget docs] : https://docs.microsoft.com/en-us/nuget/guides/create-ui-controls
Then, the "notepads" app and the "files" app can both use the "text editor" as a Nuget package - UI control.

I believe that this work would be really simple to accomplish. Though, I might add that the settings for the "Text editor" may have to either be implemented as another "Nuget package UI control" or bundled with the "text editor" UI control.
What do you think?

**Comment by JasonStein** [comment]
> Of course, putting the entire "notepads" code in the "files" app could cause lots of problems.
> 
> But I believe I MAY have found a way around it using a very simple technique:
> We could partition the "notepads" code such that the "text editor" part is separate from the "multiple tab systems" part and then, implement the "text editor" part as a Nuget package (with UI controls). Refer UWP - Nuget docs : [docs.microsoft.com/en-us/nuget/guides/create-ui-controls](https://docs.microsoft.com/en-us/nuget/guides/create-ui-controls)
> Then, the "notepads" app and the "files" app can both use the "text editor" as a Nuget package - UI control.
> 
> I believe that this work would be really simple to accomplish. Though, I might add that the settings for the "Text editor" may have to either be implemented as another "Nuget package UI control" or bundled with the "text editor" UI control.
> What do you think?

For sure, it is doable but I don't have a plan yet and it requires a lot of additional work as well. Sorry for that.

**Comment by soumyamahunt** [comment]
@enygmator Files dev has his own version of text editor app, I think if he ever feels need to integrate text editor app with his app he would rather choose his own app rather than Notepads.
